### PR TITLE
Support property aliases for resource provider API. 

### PR DESCRIPTION
**Pull request by TianoMS**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@TianoMS__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    using System.Linq;
+    using Newtonsoft.Json;
+    using Microsoft.Rest.Azure;
+
+    /// <summary>
+    /// Defines a page in Azure responses.
+    /// </summary>
+    /// <typeparam name="T">Type of the page content items</typeparam>
+    [JsonObject]
+    public class Page1<T> : IPage<T>
```
Comment (hovsepm) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs]: this looks bad. Why did we get this kind of nameing?

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator 0.14.0.0
```
Comment (hovsepm) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs]: please use 0.16 version of generator.

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs (hunk 1 above)
Comment (vivsriaus) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs]: Yes, let's remove this

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs (hunk 2 above)
Comment (vivsriaus) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs]: @TianoMS Please exclude Page1.cs from this PR.

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/SubscriptionsOperations.cs [hunk 3]
```diff
@@ -516,7 +516,7 @@ internal SubscriptionsOperations(SubscriptionClient client)
                 _responseContent = await _httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
                 try
                 {
-                    _result.Body = SafeJsonConvert.DeserializeObject<Page<Subscription>>(_responseContent, this.Client.DeserializationSettings);
+                    _result.Body = SafeJsonConvert.DeserializeObject<Page1<Subscription>>(_responseContent, this.Client.DeserializationSettings);
```
Comment (vivsriaus) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/SubscriptionsOperations.cs]: Please revert this. It should be Page<T>

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/TenantsOperations.cs [hunk 4]
```diff
@@ -186,7 +186,7 @@ internal TenantsOperations(SubscriptionClient client)
                 _responseContent = await _httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
                 try
                 {
-                    _result.Body = SafeJsonConvert.DeserializeObject<Page<TenantIdDescription>>(_responseContent, this.Client.DeserializationSettings);
+                    _result.Body = SafeJsonConvert.DeserializeObject<Page1<TenantIdDescription>>(_responseContent, this.Client.DeserializationSettings);
```
Comment (vivsriaus) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/TenantsOperations.cs]: Revert

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs (hunk 1 above)
Comment (TianoMS) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs]: This is automatically generated by AutoRest. I'll remove it manually.

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs (hunk 2 above)
Comment (TianoMS) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/Generated/Models/Page1.cs]: We'll upgrade AutoRest in a separate task.

**Pull request merged**